### DISCUSSION TITLE
Centralised configuration for home settings

### DISCUSSION THREAD

**Discussion by maia** [jmcollin78/versatile_thermostat#156] [Ideas]
Is it technically possible to have a general configuration pane with all variables that are relevant for the entire home and don't change from one room to the other? E.g. external temp sensor, cycle duration, min/max temp allowed, regulation threshold, regulation min period, window sensor delay, temperature decrease threshold, max duration of open window, presence sensor, minimal activation delay, security delay,…

This would drastically reduce the number of options when configuring another climate entity.

**Comment by jmcollin78** [comment]
"If you have money and time all is possible."

More seriously, I see the idea. Having 12 VTherm my self I know tthat when you need to change something it is a little boring.
But as far as I know all devices are autonomous and cannot communicate with each other.

If someone has the trick, it could be an interesting enhancement.

**Comment by maia** [comment]
I could imagine one would have two integrations or one addon and one integration or something like that? One of these holds the "central" configuration data, the other integration all entities. Yes, this sounds complicated. Maybe we'll discover another integration that helps finding an easy solution.

**Comment by jmcollin78** [comment]
I'm not comfortable with addon.

What seems possible simply is to have a set a configuration into the configuration.yaml. This set of configuration can be transversal to all Vtherm instances and could be replace the UI configuration set by the user.

But, you need a restart each time you change something in it (at least a restart of the integration), through the UI, you will see a value which will not be used. And this is very misleading for the user.

So I see no simple and use friendly way to achieve that.

Just wait if someone, who read this have a solution.

**Reply by adi90x** [reply]
Wouldn't it be possible in UI to either have the actual flow and maybe another "Advanced Flow" that would only by a text box in which one can input ```key: value``` lines ?
Could ease some multiple entities update, this json/yaml like config may easily be exported through a service ( and shared with other)
